rust/noosphere-gateway/src/worker/syndication.rs

```diff
@@ -42,10 +42,30 @@ pub struct SyndicationJob<C> {
 /// gives us a short-cut to determine if a block should be added.
 #[derive(Serialize, Deserialize)]
 pub struct SyndicationCheckpoint {
-    pub revision: Link<MemoIpld>,
-    pub syndicated_blocks: BloomFilter<256, 30>,
+    pub last_syndicated_version: Option<Link<MemoIpld>>,
+    pub syndication_epoch: u64,
+}
+
+impl SyndicationCheckpoint {
+    pub fn new() -> Result<Self> {
+        Ok(Self {
+            last_syndicated_version: None,
+            syndication_epoch: SystemTime::now().duration_since(UNIX_EPOCH)?.as_secs(),
+        })
+    }
+
+    pub fn lifetime(&self) -> Result<Duration> {
+        Ok(SystemTime::now().duration_since(UNIX_EPOCH)?)
```

Comment:
I think this should be the duration since `syndication_epoch`: it's unused AFAICT, and `is_expired()` will always be true (other than like, January 1970)

Reply:
Good catch this is a bug

Reply:
Fixed and added a test